Platform/ElkhartlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c

```diff
@@ -239,6 +243,45 @@ TccModePreMemConfig (
   return Status;
 }
 
+/**
+  Update S0ix flag
+
+  @param  FspmUpd            The pointer to the FSP-M UPD to be updated.
+
+  @retval none
+**/
+VOID
+UpdateS0ixStatus (
+  FSPM_UPD  *FspmUpd
+)
+{
+  FEATURES_CFG_DATA             *FeaturesCfgData;
+  PLATFORM_DATA                 *PlatformData;
+  BOOLEAN                       PchSciSupported;
+
+  PlatformData = PLAT_DATA;
+  FeaturesCfgData = (FEATURES_CFG_DATA *) FindConfigDataByTag (CDATA_FEATURES_TAG);
+  if (PlatformData == NULL || FeaturesCfgData == NULL) {
+    DEBUG ((DEBUG_INFO, "Failed to find Platform Data / Config Data! S0ix setting failed.\n"));
+    return;
+  }
+  PchSciSupported               = PchIsSciSupported ();
+  /**
+  EHL S0ix condition depends on:
+  - user turn on via config data
+  - not FUSA sku
+  - TCC config turns it on
+  **/
+  if (PchSciSupported == 1 || mTccS0ixSupport == 0 || FeaturesCfgData->Features.S0ix == 0) {

```

Comment:
Can we use PolicyConfig->Sstates instead of mTccS0ixSupport here?

Reply:
done reworked.